app/controllers/devise/two_step_verification_controller.rb

```diff
@@ -12,6 +12,13 @@ def update
     render(:show) && return if params[:code].nil?
 
     if current_user.authenticate_otp(params[:code])
+      expires_seconds = User::REMEMBER_2SV_SESSION_FOR
+      if expires_seconds && expires_seconds > 0
+        cookies.signed['remember_2sv_session'] = {
+          value: {user_id: current_user.id, expires: expires_seconds.from_now},
```

Comment:
I think the important thing to express is that isn't redundant and does need to be part of the signed value. Maybe we should have a test which tampers with the unsigned expiry?